app/src/main/java/co/minesweepers/birthday/activities/CreateMemoryActivity.java

```diff
@@ -38,19 +34,27 @@ protected void onCreate(Bundle savedInstanceState) {
         mPerson = Memory.getInstance().getPerson(personId);
     }
 
-    private void showFileChooser() {
+    private void showVideoFileChooser() {
         Intent intent = new Intent();
         intent.setType("video/*");
         intent.setAction(Intent.ACTION_GET_CONTENT);
         startActivityForResult(Intent.createChooser(intent, "Select Video"), PICK_VIDEO_REQUEST);
     }
 
+    private void showAudioFileChooser() {
+        Intent intent = new Intent();
+        intent.setType("audio/*");
+        intent.setAction(Intent.ACTION_GET_CONTENT);
+        startActivityForResult(Intent.createChooser(intent, "Select Audio"), PICK_AUDIO_REQUEST);
+    }
 
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (requestCode == PICK_VIDEO_REQUEST && resultCode == RESULT_OK && data != null && data.getData() != null) {
-            videoPath = data.getData();
+            mPerson.addVideo(data.getData());
+        } else if (requestCode == PICK_AUDIO_REQUEST && resultCode == RESULT_OK && data != null && data.getData() != null) {
```

Comment:
instead checking the `resultCode` and `data` everytime you can just check it once right at the start
```
if(resultCode == RESULT_OK && data != null && data.getData() != null) {
    switch(requestCode) {
        case: PICK_VIDEO_REQUEST:
            ....
            break;

        case: PICK_AUDIO_REQUEST:
            ....
            break;
    }
}
```